include/triton/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVMBase.h

```diff
@@ -95,6 +95,13 @@ class ElementwiseOpConversionBase : public ConvertOpToLLVMPattern<SourceOp> {
       return resultVals;
     }
 
+    // cannot utilize duplication with parent mfma layout
+    if (auto slice = encoding.dyn_cast<SliceEncodingAttr>()) {
+      if (slice.getParent().dyn_cast<AMDMfmaEncodingAttr>()) {
+        return resultVals;
+      }
+    }
```

Comment:
This looks like a workaround? Do we know why it doesn't work?

Reply:
It is related to the lowering of the `cmpi`, which is slice encoding with parent `mfma layout`
` %24 = arith.cmpi slt, %0, %22 : tensor<32xi32, #triton_gpu.slice<{dim = 1, parent = #mma}>> loc(#loc11)`

I think it is similar to the reordering of the dot operands encoding as:
https://github.com/openai/triton/blob/77195bcf28503f11c99c27253e336033d82d9fc6/lib/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVM.cpp#L28-L77

Yes, the changes are to disable the deduplication for such scenario. May be we can do the same thing for the slice encoding in the same way as dotOperand?